lang/cs/Org.Apache.REEF.Network/Group/Task/Impl/GroupCommClient.cs

```diff
@@ -84,11 +85,15 @@ public void WaitingForRegistration(CancellationTokenSource cancellationSource =
                     group.WaitingForRegistration(cancellationSource);
                 }
             }
-            catch (SystemException e)
+            catch (OperationCanceledException)
```

Comment:
Let's use

catch (Exception e) 
  if (e is OperationCanceledException || e is RemotingException) 
     Dispose(); 
  throw;

instead. By the way, why don't we dispose in other cases?